src/python_testing/matter_testing_infrastructure/chip/testing/commissioning.py

```diff
@@ -99,9 +99,28 @@ class CustomCommissioningParameters:
     randomDiscriminator: int
 
 
+class PairingStatus:
+    """
+    This class is used to store the pairing status of a commissioning process with a device.
+    if the commissioning Process is unsuccessful then we pass the exception to the class which sets the PairingStatus
+    to False. If we do not pass any exception when creating the Pairing status then the commissioning status
+    is set to true to indicate that the commissioning process has succeeded.
+    """
+
+    def __init__(self, exception: Optional[Exception] = None):
+        self.exception = exception
+        self.success = exception is None
+
+    def __bool__(self):
+        return self.success
+
+    def __str__(self):
+        return str(self.exception) if self.exception else ""
```

Comment:
Is this formatted in any way that we need empty string to work?

I would do either real human readable string:

```py
return f'PairingStatus<exception:{self.exception}>'
```

or consider writing out more than empty:

```py
return str(self.exception) if self.exception else "<PairingSuccess>"
```

or similar.